### ISSUE TITLE
open hint in specified container otherwise create it

### ISSUE THREAD

**Issue by forgetmenotz** [tridactyl/tridactyl#1996]
Hi, 

I am stuck on this problem in tridactyl, I was wondering if anyone could help me, I have read :help and example rcs.

The primary aim is to put a hinted url into the cmdline so it looks like this `:tabopen https://someurl.com -c`

First I try to get the url alone:
I can send a hint to alert box like so: 
`composite hint -y | js -p alert(JS_ARG)`
I can send a string to fillcmdline:
`js let myvar="sometext" ; fillcmdline(myvar)`
But I tried to store the url in a variable
`composite hint -y | js -p let myvar=JS_ARG`

...which seems to fail, or (at least) tacking `; fillcmdline(myvar)` on the end does. I know the help mentioned pipes and semicolons. But I'm not sure how I can make use of alias here. I was also unsuccessful trying to use getclip.

Be super grateful if anyone has any pointers. I am a novice.

**TLDR**

Is it possible to program a command to yank a chosen hint then open the cmdline with `:newtab https://hintedurlhere -c` ?

**Comment by glacambre** [comment]
Hi, does the following command do what you want?
```
composite hint -y | js -p tri.excmds.fillcmdline(`tabopen ${JS_ARG.href ? JS_ARG.href : JS_ARG.textContent} -c`)
```

**Comment by forgetmenotz** [comment]
Wow, thanks for the quick response. 

I didn't have time to edit the post, it seems I missed an important factor here that -c (and the container name) seems to be required before the url, meaning that the cursor would have to be somehow positioned after it, and before the url. But your solution definitely did exactly what I asked originally.

**Comment by glacambre** [comment]
```
composite hint -y | js -p tri.excmds.fillcmdline(`tabopen -c ${JS_ARG.href ? JS_ARG.href : JS_ARG.textContent}`) ; sleep 20 ; text.beginning_of_line ; text.forward_word ; text.forward_word
```
Will probably do what you want then. I'm not sure the `sleep` command is needed but I put it there just in case.

**Comment by forgetmenotz** [comment]
Yes that works. I did try the tri.excmd things. But it looks like I didn't know the js. The only other thing would be to create a container if it doesn't exist. Maybe I can do it. Thanks for the solution!

**Comment by glacambre** [comment]
Creating a container if doesn't exist will require using mozilla's [contextualIdentities](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contextualIdentities) api, which resides in the background script (so you'll need to use `jsb` instead of `js`).

**Comment by bovine3dom** [comment]
I'd quite like "automatically make containers" to be a setting built in to Tridactyl. We already have `auconcreatecontainer`, should just be a case of copying and that logic to tabopen and then editing it a bit to add another setting.

Would you be interested in doing that, @forgetmenotz?

**Comment by forgetmenotz** [comment]
Hey thanks for the offer to contribute to this awesome addon. I do find something like this interesting. I could probably dig around finding the bits that create the container and try to merge them with the fuzzy matching part and stick it in the tabopen part. But I am a newbie here. I haven't learned to use github yet for example. I have done similar tinkering with other software before...I have aspirations to learn to code...yet I must confess find it hard to comprehend the inner workings of big software programs like this and firefox. And that has held me back from trying things, so I get distracted perfecting my vim, tmux, i3 setup etc. I'll let you know if I get the "fuzzy container search or create" working though. Cheers.

**Comment by bovine3dom** [comment]
We'd be happy to help. Just pop on to Riot :)
